From 7d11e75581640a79014d41532c570e4a5d7b2a62 Mon Sep 17 00:00:00 2001
From: Liang Dai <liang.dai1@citrix.com>
Date: Mon, 2 Jul 2018 19:31:30 +0800
Subject: [PATCH] [PATCH] Revert "vgcreate, pvcreate, vgextend: don't use a
 device with  duplicates

This reverts commit 86f92714574487cce6a9de785ba2ff93b23a98f7.

There is a scenario in a enviroment with remote storage and multiple
paths should work.
diff --git a/lib/cache/lvmcache.c b/lib/cache/lvmcache.c
index f55a14c..d135967 100644
--- a/lib/cache/lvmcache.c
+++ b/lib/cache/lvmcache.c
@@ -1788,17 +1788,6 @@ const char *lvmcache_pvid_from_devname(struct cmd_context *cmd,
 	return dev->pvid;
 }
 
-int lvmcache_pvid_in_unchosen_duplicates(const char *pvid)
-{
-	struct device_list *devl;
-
-	dm_list_iterate_items(devl, &_unused_duplicate_devs) {
-		if (!strncmp(devl->dev->pvid, pvid, ID_LEN))
-			return 1;
-	}
-	return 0;
-}
-
 static int _free_vginfo(struct lvmcache_vginfo *vginfo)
 {
 	struct lvmcache_vginfo *primary_vginfo, *vginfo2;
diff --git a/lib/cache/lvmcache.h b/lib/cache/lvmcache.h
index bf976e9..d515ee7 100644
--- a/lib/cache/lvmcache.h
+++ b/lib/cache/lvmcache.h
@@ -207,8 +207,6 @@ int lvmcache_dev_is_unchosen_duplicate(struct device *dev);
 
 void lvmcache_remove_unchosen_duplicate(struct device *dev);
 
-int lvmcache_pvid_in_unchosen_duplicates(const char *pvid);
-
 int lvmcache_get_vg_devs(struct cmd_context *cmd,
 			 struct lvmcache_vginfo *vginfo,
 			 struct dm_list *devs);
diff --git a/tools/toollib.c b/tools/toollib.c
index e1c86f9..bab9f70 100644
--- a/tools/toollib.c
+++ b/tools/toollib.c
@@ -5097,15 +5097,6 @@ static int _pvcreate_check_single(struct cmd_context *cmd,
 
 
 	/*
-	 * Don't allow using a device with duplicates.
-	 */
-	if (lvmcache_pvid_in_unchosen_duplicates(pd->dev->pvid)) {
-		log_error("Cannot use device %s with duplicates.", pd->name);
-		dm_list_move(&pp->arg_fail, &pd->list);
-		return 1;
-	}
-
-	/*
 	 * What kind of device is this: an orphan PV, an uninitialized/unused
 	 * device, a PV used in a VG.
 	 */
